### PR TITLE
Pleiades primary

### PR DESCRIPTION
**Pull request by nekhtiari**
Pull Request
============

> Tracked in JIRA by [IN-XXX](https://geoinformationstore.atlassian.net/browse/IN-XXX)
> Paired with @a, @b

### Scope of the PR

Describe the scope of this pull request. Is this a new feature or a bug fix?

### Checklist

- [x] Checked spelling and grammar
- [x] Provided code samples where relevant
- [x] Checked the output of `make html` to ensure new content displays correctly

### CODE REVIEW

File: source/up42-blocks/data/pleiades-primary.rst [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+
+The available output bands of the Pleiades block are: panchromatic, red, green, blue and near infrared. Pansharpened Pleiades HR imagery has a spatial resolution of 0.5x0.5m. The multispectral bands have a resolution of 2x2m.
+
+.. tip::
+
+   Find out more about the RPCs `here <https://gis.stackexchange.com/questions/180414/how-rational-polynomial-coefficientsrpcs-are-calculated-need-references>`_. Unprojected Pleiades Primary image can be easily orthorectified using RPC and GDAL command as follows: `gdalwarp -rpc input_image output_image`. For more information look at `here <https://gdal.org/programs/gdalwarp.html>`_.
```
Comment (markusuwe) [inline source/up42-blocks/data/pleiades-primary.rst]: Nice!

File: source/up42-blocks/data/pleiades-primary.rst [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+* ``clip_to_aoi``: When set to ``true``, the :term:`AOI` specified via ``bbox``, or ``contains``, or ``intersect`` for the previous data block in the workflow will **first** be clipped and then processed. Please note that by **default** this parameter is set to ``true``.
+* ``stereo_images_only``: When set to ``true``, based on the :term:`AOI` specified via ``bbox``, or ``contains``, or ``intersect`` and chosen time, all **possible** (tri-)stereo images will be returned. If these these images exist, they are usually captured in less than a minute with almost equal coverage (for more information on almost equals definition look `here <https://shapely.readthedocs.io/en/latest/manual.html>`_).
+
+.. note::
+
+  Please when using ``stereo_images_only`` filter, set the ``limit`` to the maximum number, therefore all available images for the certain AOI will be checked. Also, due to the scarcity of (tri-) steroe images, it is highly **recommended** to do ``dry_run`` to make sure if any stereo images exist to avoid any additional cost.
```
Comment (markusuwe) [inline source/up42-blocks/data/pleiades-primary.rst]: high number?

File: source/up42-blocks/data/pleiades-primary.rst [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+
+The available output bands of the Pleiades block are: panchromatic, red, green, blue and near infrared. Pansharpened Pleiades HR imagery has a spatial resolution of 0.5x0.5m. The multispectral bands have a resolution of 2x2m.
+
+.. tip::
+
+   Find out more about the RPCs `here <https://gis.stackexchange.com/questions/180414/how-rational-polynomial-coefficientsrpcs-are-calculated-need-references>`_. Unprojected Pleiades Primary image can be easily orthorectified using RPC and GDAL command as follows: `gdalwarp -rpc input_image output_image`. For more information look at `here <https://gdal.org/programs/gdalwarp.html>`_.
```
Comment (perusio) [inline source/up42-blocks/data/pleiades-primary.rst]: Can be easily orthorectified using the ``gdalwarp`` command line tool from GDAL. Example:


```bash
   gdal_warp -rpc <input image> <output image>
```

Please consult the ``gdalwarp`` `documentation <https://gdal.org/programs/gdalwarp.html>`_ for better understanding  on how  to use it.

File: source/up42-blocks/data/pleiades-primary.rst [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+* ``clip_to_aoi``: When set to ``true``, the :term:`AOI` specified via ``bbox``, or ``contains``, or ``intersect`` in the workflow will be clipped from the scene. Please note that by **default** this parameter is set to ``true``.
+* ``stereo_images_only``: When set to ``true``, based on the :term:`AOI` specified via ``bbox``, or ``contains``, or ``intersect`` and chosen time, all **possible** (tri-)stereo images will be returned. If these these images exist, they are usually captured in less than a minute with almost equal coverage (for more information on almost equals definition look `here <https://shapely.readthedocs.io/en/latest/manual.html>`_).
+
+.. note::
+
+  Please when using ``stereo_images_only`` filter, set the ``limit`` to the higher number, therefore all available images for the certain AOI will be checked. Also, due to the scarcity of (tri-) steroe images, it is highly **recommended** to do ``dry_run`` to make sure if any stereo images exist to avoid any additional cost.
```
Comment (perusio) [inline source/up42-blocks/data/pleiades-primary.rst]: Please bear in mind that when setting `stereo_images_only` to `true` you should also set the `limit` parameter to highest possible value (`500`) to make sure to search for stereo imagery among the largest possible archive.

```rst
.. tip::

    When trying to obtain stereo/tri-stereo images you should **always** perform a :term:`TestQuery` to make sure that there are available stereo and/or tri-stereo images, thus avoiding to incur costs when retrieving the imagery.
```

File: source/up42-blocks/data/pleiades-primary.rst [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+
+.. warning::
+
+  Due to nature of un-rectified imagery, applying ``clip_to_aoi`` can cause a drastic shift in the position of the clipped image (After applying orthorectification). We provided a mechanism to overcome this issue. However, the drawback of this approach is that the output AOI will be an approximation of the input AOI that includes a buffer.
```
Comment (perusio) [inline source/up42-blocks/data/pleiades-primary.rst]: Due to nature of un-rectified imagery, applying ``clip_to_aoi`` can cause a substantial shift in the position of the clipped image &mdash; after applying orthorectification. We provide a mechanism to overcome this issue. However, this mechanism has as side-effect that the output AOI will be an *approximation* of the input AOI: it  includes a buffer region.

File: source/up42-blocks/data/pleiades-primary.rst [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+	"time": "2018-01-01T00:00:00+00:00/2020-12-31T23:59:59+00:00",
+	"limit": 1,
+	"zoom_level": 18,
+	"clip_to_aoi": true,
+	"max_cloud_cover": 100,
+	"stereo_images_only": false
+	}
+  }
+
+Example query using ``stereo_images_only`` in ``dry_run`` mode. Please **note** that in this query the ``limit`` set to the **maximum**:
```
Comment (perusio) [inline source/up42-blocks/data/pleiades-primary.rst]: Searching for query stereo and tri-stereo images using ``stereo_images_only`` with a  :term:`TestQuery` ( ``DRY_RUN``  mode). Pay particular attention that the value  of `limit` is set to its maximum (500).

File: source/up42-blocks/data/pleiades-primary.rst [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+		"spectralRange": "VISIBLE",
+		"title": "DS_PHR1B_201810161038295_FR1_PX_E013N52_0513_01724",
+		"workspaceId": "0e33eb50-3404-48ad-b835-b0b4b72a5625",
+		"workspaceName": "public",
+		"workspaceTitle": "Public"
+	  }
+	}
+  ]
+  }
+For this query, three stereo images at following acquisition date are returned: ``"2018-10-16T10:39:06.555Z"``, ``"2018-10-16T10:39:26.181Z"``, and ``"2018-10-16T10:39:43.431Z"``. They were captured in less than a minute and cover almost equal area.
```
Comment (perusio) [inline source/up42-blocks/data/pleiades-primary.rst]: at the following acquisition dates ``"2018-10-16T10:39:06.555Z"``, ``"2018-10-16T10:39:26.181Z"``, and ``"2018-10-16T10:39:43.431Z"``. They were captured within less than a minute and each covers an almost equal surface area.

File: source/up42-blocks/data/pleiades-primary.rst [hunk 8]
```diff
@@ -10,13 +10,22 @@ Please see the `block details page <https://marketplace.up42.dev/block/d1e5e0de-
 
 Block type: ``DATA``
 
-This block provides **un-rectified** Pléiades HR imagery with `Rational Polynomial Coefficients (RPCs) <https://en.wikipedia.org/wiki/Rational_polynomial_coefficient>`_. The block outputs a GeoTIFF file and can be the entire Pleiades scene or clipped to the AOI. This block can be used mainly by experts who either want to apply their own `orthorectification <https://trac.osgeo.org/ossim/wiki/orthorectification>`_ or create a 3D model based on the (tri-)stereo images (for more information look at the `Stereophotogrammetry <https://en.wikipedia.org/wiki/Photogrammetry>`_).
+This block provides **un-rectified** Pléiades HR imagery with `Rational Polynomial Coefficients (RPCs) <https://en.wikipedia.org/wiki/Rational_polynomial_coefficient>`_. The block outputs a GeoTIFF file and can be the entire Pleiades scene or clipped to the AOI. This is block to be used mainly by experts by who either want to apply their own `orthorectification <https://trac.osgeo.org/ossim/wiki/orthorectification>`_ procedure or create a tridimensional model based on stereo/tri-stereo images. Further information on `Stereophotogrammetry <https://en.wikipedia.org/wiki/Photogrammetry>`_.
```
Comment (perusio) [inline source/up42-blocks/data/pleiades-primary.rst]: This block is to be used mainly by experts

File: source/up42-blocks/data/pleiades-primary.rst [hunk 9]
```diff
@@ -10,13 +10,22 @@ Please see the `block details page <https://marketplace.up42.dev/block/d1e5e0de-
...
 
-The available output bands of the Pleiades block are: panchromatic, red, green, blue and near infrared. Pansharpened Pleiades HR imagery has a spatial resolution of 0.5x0.5m. The multispectral bands have a resolution of 2x2m.
+The available output bands of the Pleiades block are: Panchromatic, Red, Green, Blue (RGB) and Near InfraRed (NIR).. Pansharpened Pleiades HR imagery has a spatial resolution of 0.5x0.5m. The multispectral bands have a resolution of 2x2m.
```
Comment (perusio) [inline source/up42-blocks/data/pleiades-primary.rst]: Double period at the end of the sentence.
> (NIR).

File: source/up42-blocks/data/pleiades-primary.rst [hunk 10]
```diff
@@ -39,11 +48,15 @@ For more information on supported filters, see :ref:`query filter section  <filt
 
 .. note::
 
-  Please when using ``stereo_images_only`` filter, set the ``limit`` to the higher number, therefore all available images for the certain AOI will be checked. Also, due to the scarcity of (tri-) steroe images, it is highly **recommended** to do ``dry_run`` to make sure if any stereo images exist to avoid any additional cost.
+  Please bear in mind that when setting ``stereo_images_only`` to ``true`` you should also set the ``limit`` parameter to highest possible value ``(500)`` to make sure to search for stereo imagery among the largest possible archive.
+
```
Comment (perusio) [inline source/up42-blocks/data/pleiades-primary.rst]: hmm,
> archive set
reads better, yeah I know I should have suggested it before.

**Pull request merged**